### ISSUE TITLE
Allow disabling Kubernetes

### ISSUE THREAD

**Issue by Maddin-619** [rancher-sandbox/rancher-desktop#985]
### Preflight Checklist

- [X] I have searched the [issue tracker](https://github.com/rancher-sandbox/rancher-desktop/issues) for a feature request that matches the one I want to file, without success.

### Problem Description

The kubernetes single-node cluster always gets deployed when launching RD. But I don't always need to have k8s running especially when I have an external cluster. What I always need is containerd/dockerd for building/testing container images with the docker CLI or nerdctl. In summary, this means that resources are consumed for something that I do not necessarily need to run at least not for the whole time.

### Proposed Solution

In Docker Desktop there is a checkbox to enable/disable the kubernetes single-node cluster when starting Docker Desktop: 
![image](https://user-images.githubusercontent.com/19187483/142854822-ad6b06dc-54ad-4968-acee-2b5560a1f68d.png)

I think this feature would be great to see in RD too for saving system resources.

**Comment by rfay** [comment]
The title here could be improved, maybe "Allow disabling Kubernetes for those who don't need it"

**Comment by n0rig** [comment]
@Maddin-619 Can you update the feature request title

**Comment by n0rig** [comment]
fyi:
> It looks like these steps should be made optional:
> 
> https://github.com/rancher-sandbox/rancher-desktop/blob/a1caf74cf57a2aaac2e05ee4040d904e0fd1c76e/src/k8s-engine/wsl.ts#L1004
> 
> 
> https://github.com/rancher-sandbox/rancher-desktop/blob/a1caf74cf57a2aaac2e05ee4040d904e0fd1c76e/src/k8s-engine/lima.ts#L1146

**Comment by avodaqstephan** [comment]
This would be awesome and would probably save a lot of overhead. 
Right now, Rancher Desktop goes pretty ham on the CPU resources (ofc it depdens on the PC you are running it on). That's the only thing which is currently stopping us from moving the whole department to Rancher Desktop. 

I'd assume that it's possible to bring down the minimum CPU requirement from 2 to 1 CPUs in the settings. E.g. on my i5 MacBook Pro, "qemu" + "rancher desktop" are hovering around 15-23% in idle. 
Just running Lima is around 1-3%.

**Comment by n0rig** [comment]
If you want to “disable” kubernetes in rancher-desktop use these commands:

```shell
kubectl config use-context rancher-desktop
kubectl delete node lima-rancher-desktop
```

![image](https://user-images.githubusercontent.com/1235638/149979506-54cff6c3-54e4-4e84-9163-c755d720c0fc.png)
![image](https://user-images.githubusercontent.com/1235638/149979532-8aa8b868-fac2-4fd9-8b52-e7a3d1076d77.png)

**Comment by jandubois** [comment]
@n0rig This "workaround" didn't work for me. I tried both containerd and moby (on macOS), and in either case, all pods remained running ager deleting the node. Did you do anything else to get the pods into `pending` state?

In a way I didn't expect this to work: the containers are all re-parented to PID 1, so they are not stopped even when you stop k3s. That way you can do zero-downtime updates without affecting your workloads. So I'm wondering why your pods have been removed.

**Comment by Clockwork757** [comment]
I deleted the node but still noticed high cpu usage from qemu. I figured out you can stop the k3s service inside lima pretty easily:

```sh
LIMA_HOME="$HOME/Library/Application Support/rancher-desktop/lima" "/Applications/Rancher Desktop.app/Contents/Resources/resources/darwin/lima/bin/limactl" shell 0

# In the vm shell

sudo rc-service k3s stop
```

It might leave some pod containers running so you'll have to manually kill them with `docker/nerdctl kill`.

I haven't noticed any issues using docker afterwards. I would still appreciate a way to just turn off kubernetes.

**Comment by avodaqstephan** [comment]
Doesn't work for me. k3s is stopping but nerdctl is not working anymore.

I get the following error afterwards:
`FATA[0000] failed to dial "/run/k3s/containerd/containerd.sock": connection error: desc = "transport: error while dialing: dial unix /run/k3s/containerd/containerd.sock: connect: connection refused"` 

But by stopping k3s the cpu usage of qemu drops from 45-50% to less than 10%. We really need the feature to turn off Kubernetes.

**Comment by A-Shevchenko** [comment]
FYI, disabling Kubernetes by removing node (as suggested in FAQ) breaks DNS resolution in containers run by `docker run` (dockerd runtime).

**Comment by gaktive** [comment]
Spike to see if we can get this in for our Next release: #1559

**Comment by ericpromislow** [comment]
Work is being done in #1562

**Comment by jandubois** [comment]
This has been implemented in #1562 and will be in the upcoming 1.1.0 release.

We are not going to investigate the issue with "the workaround", namely deleting the node breaking DNS. That is not a supported use-case.
